akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -206,7 +227,9 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
           replicator ! Replicator.Update(ddataKey, EmptyORMultiMap, settings.writeConsistency) { registry ⇒
             ServiceRegistry(registry).removeBinding(key, entry).toORMultiMap
           }
-          Behaviors.same
+          // tombstone removals so they are not re-added by merging with other concurrent
+          // registrations for the same key
+          next(newState = registry.addTombstone(serviceInstance, setup.newTombstoneDeadline()))
```

Comment:
if we ever add `UnRegister` message in addition to only remove terminated we would have to keep track of tombstones per service key, since an ActorRef can be registered for multiple keys and only unregistered for some of them

worth adding a comment about this?

Reply:
Nah, I think adding that feature will require some additional careful thinking and looking at the terminated logic anyway.

Reply:
ok